pkg/controller/v1alpha1/inferencegraph/controller.go

```diff
@@ -173,9 +178,50 @@ func (r *InferenceGraphReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return reconcile.Result{}, errors.Wrapf(err, "fails to create DeployConfig")
 	}
 
+	// examine DeletionTimestamp to determine if object is under deletion
+	if graph.ObjectMeta.DeletionTimestamp.IsZero() {
+		// The object is not being deleted, so if it does not have our finalizer,
+		// then lets add the finalizer.
+		if !utils.Includes(graph.ObjectMeta.Finalizers, constants.InferenceGraphFinalizerName) {
+			graph.ObjectMeta.Finalizers = append(graph.ObjectMeta.Finalizers, constants.InferenceGraphFinalizerName)
+			patchYaml := "metadata:\n  finalizers: [" + strings.Join(graph.ObjectMeta.Finalizers, ",") + "]"
+			patchJson, _ := yaml.YAMLToJSON([]byte(patchYaml))
+			if err = r.Patch(ctx, graph, client.RawPatch(types.MergePatchType, patchJson)); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+	} else {
+		// The object is being deleted
+		if utils.Includes(graph.ObjectMeta.Finalizers, constants.InferenceGraphFinalizerName) {
+			// our finalizer is present, so lets handle any external dependency
```

Comment:
I am trying to follow the code and have a question about this .. external dependency.
What kind of external dependency are you referring to here?

Reply:
Uhm... No external dependency, I think.
It was copy-pasted from ISVC controller: https://github.com/opendatahub-io/kserve/blob/a009547a88cf361c371c1ba2ff51446f4954518d/pkg/controller/v1beta1/inferenceservice/controller.go#L153.

I'll remove the comment :-)

Reply:
I changed the comment.